src/Microsoft.Tye.Core/PublishProjectStep.cs

```diff
@@ -36,18 +37,29 @@ public override async Task ExecuteAsync(OutputContext output, ApplicationBuilder
 
             output.WriteDebugLine("Running 'dotnet publish'.");
             output.WriteCommandLine("dotnet", $"publish \"{project.ProjectFile.FullName}\" -c Release -o \"{outputDirectory.DirectoryPath}\"");
-            var capture = output.Capture();
-            var exitCode = await Process.ExecuteAsync(
+
+            var publishResult = await ProcessUtil.RunAsync(
                 $"dotnet",
                 $"publish \"{project.ProjectFile.FullName}\" -c Release -o \"{outputDirectory.DirectoryPath}\"",
                 project.ProjectFile.DirectoryName,
-                stdOut: capture.StdOut,
-                stdErr: capture.StdErr);
+                throwOnError: false);
 
-            output.WriteDebugLine($"Done running 'dotnet publish' exit code: {exitCode}");
-            if (exitCode != 0)
+            output.WriteDebugLine($"Done running 'dotnet publish' exit code: {publishResult.ExitCode}");
+            if (publishResult.ExitCode != 0)
             {
                 outputDirectory.Dispose();
+                output.WriteInfoLine($"'dotnet publish' failed. Error:");
+
+                foreach (var line in publishResult.StandardOutput.Split(Environment.NewLine))
+                {
+                    output.WriteInfoLine(line);
+                }
+
+                foreach (var line in publishResult.StandardError.Split(Environment.NewLine))
+                {
+                    output.WriteInfoLine(line);
+                }
```

Comment:
This will be fine, but it wasn't really what I had in mind. I'll send a follow up :)

Reply:
Yeah I had a feeling it wasn't.